### PR TITLE
[MM-27330] Add some missing search related actions

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

PR adds some missing search related actions to `SimulController`.

#### Ticket 

https://mattermost.atlassian.net/browse/MM-27330

### CODE REVIEW

File: loadtest/control/simulcontroller/actions.go [hunk 1]
```diff
@@ -870,3 +870,95 @@ func searchChannels(u user.User) control.UserActionResponse {
...
+	if rand.Float64() < 0.2 {
+		channel, err := u.Store().RandomChannel(team.Id, store.SelectMemberOf|store.SelectNotDirect|store.SelectNotGroup)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		opts.In = channel.Name
+	}
+
+	if rand.Float64() < 0.2 {
+		opts.On = time.Now()
```
[the remark this answers is not in the document]
Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go:906]: Yes. Let me update.

[the remark this answers is not in the document]
Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go:906]: Regarding the older date. The resolution of the search appears to be a day so unless the load-test has been running for a long time it shouldn't make much of a difference.

File: loadtest/control/simulcontroller/actions.go [hunk 2]
```diff
@@ -870,3 +870,101 @@ func searchChannels(u user.User) control.UserActionResponse {
...
+	})
+}
+
+func searchGroupChannels(u user.User) control.UserActionResponse {
+	user, err := u.Store().RandomUser()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.EmulateUserTyping(user.Username[:1+rand.Intn(4)], func(term string) control.UserActionResponse {
```
Comment (agnivade) [inline loadtest/control/simulcontroller/actions.go:961]: Let's add the comment about 4 here also for clarity

File: loadtest/control/simulcontroller/actions.go [hunk 3]
```diff
@@ -870,3 +870,101 @@ func searchChannels(u user.User) control.UserActionResponse {
...
+func searchUsers(u user.User) control.UserActionResponse {
+	user, err := u.Store().RandomUser()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	return control.EmulateUserTyping(user.Username, func(term string) control.UserActionResponse {
+		users, err := u.SearchUsers(&model.UserSearch{
+			Term:  term,
+			Limit: 100,
```
Comment (agnivade) [inline loadtest/control/simulcontroller/actions.go:946]: Is this the limit used by the webapp?

File: loadtest/control/simulcontroller/actions.go [hunk 4]
```diff
@@ -870,3 +870,101 @@ func searchChannels(u user.User) control.UserActionResponse {
...
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	var words []string
+	var opts control.PostsSearchOpts
+	// This is an arbitrary limit on the number of words to search for.
+	// TODO: possibly use user analytics data to improve this.
+	count := 1 + rand.Intn(4)
+
+	// TODO: back the probability of these choices with real data.
```
Comment (agnivade) [inline loadtest/control/simulcontroller/actions.go:888]: Let's file a ticket for this? Now that you have access, it shouldn't be very hard to get this.

File: loadtest/control/simulcontroller/actions.go [hunk 5]
```diff
@@ -870,3 +870,101 @@ func searchChannels(u user.User) control.UserActionResponse {
...
+	}
+
+	timeOpts := []*time.Time{
+		&opts.On,
+		&opts.Before,
+		&opts.After,
+	}
+	if rand.Float64() < 0.2 {
+		to := timeOpts[rand.Intn(len(timeOpts))]
+		*to = time.Now().Add(-time.Duration(rand.Intn(7)) * time.Hour * 24)
```
Comment (agnivade) [inline loadtest/control/simulcontroller/actions.go:912]: Why 7?

And also, this seems like a roundabout way to do things. Is there a reason we cannot just get a random number between 0 to 3 and update the right variable accordingly?

File: loadtest/control/simulcontroller/actions.go (hunk 4 above)
Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go:888]: I am not sure we collect this kind of data. To improve this we would need to store the actual text content of what the users are searching.
Were you thinking of something else?

File: loadtest/control/simulcontroller/actions.go (hunk 5 above)
Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go]: > Why 7?

It's an arbitrary value like everything else in this kind of methods. I chose to go back not more than a week. hence the 7. This was to address @isacikgoz's review.

> And also, this seems like a roundabout way to do things. Is there a reason we cannot just get a random number between 0 to 3 and update the right variable accordingly?

It's the first thing I thought of. Maybe it was the [C](https://en.wikipedia.org/wiki/C_(programming_language)) coder inside me. I *simplified*.

File: loadtest/control/simulcontroller/actions.go (hunk 3 above)
Comment (streamer45) [reply loadtest/control/simulcontroller/actions.go:969]: Yes.

**Comment by streamer45** [comment]
> One concern I had is that we are not emulating the autocomplete of users and channels when the user is actually typing the "in:" or "from:" string. That also generates some autocomplete queries which we should do. Perhaps a separate ticket is appropriate for this?

That's a good point. I added some logic to make it a bit more realistic.

File: loadtest/control/simulcontroller/actions.go (hunk 4 above)
Comment (agnivade) [reply loadtest/control/simulcontroller/actions.go:888]: Oh yeah you're right. Nvm then.

**Pull request merged**